addons/purchase_stock/models/purchase.py

```diff
@@ -333,6 +333,10 @@ def write(self, values):
         lines = self.filtered(lambda l: l.order_id.state == 'purchase')
         previous_product_qty = {line.id: line.product_uom_qty for line in lines}
         result = super(PurchaseOrderLine, self).write(values)
+        if 'price_unit' in values:
+            for line in lines:
+                moves = line.move_ids.filtered(lambda s: s.state not in ('cancel', 'done'))
+                moves.write({'price_unit': line._get_stock_move_price_unit()})
```

Comment:
What about moves from kit components? Wouldn't this write a wrong price on them?

Reply:
as far as I understand kit price does not work properly for purchase independently from this PR because the components stock.move's price_unit is always zero even if the PO line price is not.

Reply:
In any case the don't get the price of the purchase line. This is the safeguard for it:

- The stock moves in the picking for the components don't get the price unit from the one the are exploded from: https://github.com/odoo/odoo/blob/5df214b1a8160ccdfa73b926f1257d5a41de537c/addons/stock/models/stock_move.py#L117
- They never get valued at their linked purchase line price unit: https://github.com/odoo/odoo/blob/5df214b1a8160ccdfa73b926f1257d5a41de537c/addons/purchase_stock/models/stock.py#L40

But this change is different as they're being written directly.

Reply:
You're right. We should add a `s.product_id == line.product_id` in the lambda to avoid updating kit components moves.

Reply:
Will you open a PR for it?

Reply:
done : https://github.com/odoo/odoo/pull/97503